### ISSUE TITLE
ExploreAngular - Ch16 - Page635 - TS2.7 - strictPropertyInitialization

### ISSUE THREAD

**Issue by NJFfromFrance** [markjprice/cs7dotnetcore#4]
TypeScript 2.7 introduced strictPropertyInitialization which breaks 'home.component.ts' in the ExporeAngular example in Ch16, page 635, as 'customers' generates this error: [ts] Property 'customers' has no initializer and is not definitely assigned in the constructor.

Fix: add: "strictPropertyInitialization": false, to tsconfig.json and restart VSCode.

**Comment by markjprice** [comment]
Thanks for spotting this. I've updated the tsconfig.json files in the correct repository: cs7dotnetcore2

**Comment by NJFfromFrance** [comment]
Hi Mark,

 

You are welcome.

 

Thank you for writing this book. I’ve learned a lot from it.

I have used it to learn myself how to program against dotnetcore in C# using VSCode (on Windows).

C# I already knew quite well, though not the latest features.

 

I have noticed quite a few errors, mistakes, typos in the text of the book.

Consulting the apparently frequently updated source code on GitHub, has saved me.

Thank you for that. PacktPub has alerted me to these updates.

 

BTW, I think that VSCode/C#/dotnetcore is a great environment with not to many books/resources.

I could think of a book on unit testing in this environment or maybe even about writing a db backend for your website using SQL Server or MongoDB.

Just my thoughts.

 

Cheers, Niek Jan Ferwerda (niekjan.ferwerda@gmail.com)            

(from the Netherlands, now in France, also lived in the UK).

 

Van: Mark J Price [mailto:notifications@github.com] 
Verzonden: zaterdag 24 februari 2018 10:21
Aan: markjprice/cs7dotnetcore <cs7dotnetcore@noreply.github.com>
CC: Niek Jan Ferwerda <niekjan.ferwerda@gmail.com>; Author <author@noreply.github.com>
Onderwerp: Re: [markjprice/cs7dotnetcore] ExploreAngular - Ch16 - Page635 - TS2.7 - strictPropertyInitialization (#4)

 

Thanks for spotting this. I've updated the tsconfig.json files in the correct repository: cs7dotnetcore2
